### PR TITLE
Initial work on client interface.

### PR DESCRIPTION
**Pull request by Fraser999**
Renamed `interface` to `node_interface`.

[Review on Reviewable](https://reviewable.io/reviews/dirvine/routing/110)

### CODE REVIEW

**Comment by dirvine** [comment]
Reviewed files:
- src/interface.rs @ r1
- src/lib.rs @ r1
- src/node_interface.rs @ r1
- src/routing_client.rs @ r1
- src/routing_node.rs @ r1

---

<sup>**[src/routing_client.rs, line 202 \[r1\]](https://reviewable.io:443/reviews/dirvine/routing/110#-JnYUC7FV-417o-6rl_b)** ([raw file](https://github.com/dirvine/routing/blob/505eb4465816a7edead795a76edc2d457164af28/src/routing_client.rs#L202)):</sup>
reminder

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/routing/110)

File: src/routing_client.rs [hunk 1]
```diff
@@ -220,9 +199,13 @@ impl<'a, F> RoutingClient<'a, F> where F: Interface {
     }
 
     /// Add something to the network, will always go via ClientManager group
+<<<<<<< Updated upstream
```
Comment (Viv-Rajkumar) [inline src/routing_client.rs]: why are we having unresolved conflicts?

Comment (Fraser999) [inline src/routing_client.rs]: Cos git sucks :)

File: src/lib.rs [hunk 2]
```diff
@@ -48,32 +48,37 @@
 #![allow(dead_code, unused_variables, unused_features, unused_attributes)]
 #![feature(custom_derive, rand, collection, std_misc, unsafe_destructor, unboxed_closures, io, core,
            thread_sleep, ip_addr, convert, scoped)]
```
Comment (rossmuir) [inline src/lib.rs]: Can we add `#![deny(missing_docs)]`

**Comment by dirvine** [comment]
Reviewed files:
- Cargo.toml @ r3
- src/client_interface.rs @ r1
- src/macros.rs @ r3
- src/message_interface.rs @ r3
- src/name_type.rs @ r3
- src/sendable.rs @ r3

---

<sup>**[Cargo.toml, line 4 \[r3\]](https://reviewable.io:443/reviews/dirvine/routing/110#-JnayJLIxNB9V_VRo2i6)** ([raw file](https://github.com/dirvine/routing/blob/acd6e86693f4698c30caa194e6994a813366d5f0/Cargo.toml#L4)):</sup>
Can this be just 0.1.2 as that has never been published or used by any other lib. 

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/routing/110)

**Pull request merged**